services/graph/pkg/validate/libregraph.go

```diff
@@ -0,0 +1,76 @@
+package validate
+
+import (
+	"github.com/go-playground/validator/v10"
+	libregraph "github.com/owncloud/libre-graph-api-go"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/owncloud/ocis/v2/services/graph/pkg/unifiedrole"
+)
+
+// initLibregraph initializes libregraph validation
+func initLibregraph(v *validator.Validate) {
+	driveItemInvite(v)
+}
+
+// driveItemInvite validates libregraph.DriveItemInvite
+func driveItemInvite(v *validator.Validate) {
+	s := libregraph.DriveItemInvite{}
+
+	v.RegisterStructValidationMapRules(map[string]string{
+		"Recipients":         "min=1",
+		"Roles":              "max=1",
+		"ExpirationDateTime": "omitnil,gt",
+	}, s)
+
+	v.RegisterStructValidation(func(sl validator.StructLevel) {
+		driveItemInvite := sl.Current().Interface().(libregraph.DriveItemInvite)
+
+		switch {
+		case len(driveItemInvite.Roles) == len(driveItemInvite.LibreGraphPermissionsActions):
```

Comment:
This seems wrong. It only catches the case when there is exactly on role and exactly one item in the `permission.actions`. It will happily accept a request like this:

```
{
	"recipients": [
		{
			"objectId": "058bff95-6708-4fe5-91e4-9ea3d377588b"
		}
	],
	"roles": [
		"b1e2218d-eef8-4d4c-b82d-0f1a1b48f3b5"
	],
	"@libre.graph.permissions.actions": [
		"libre.graph/driveItem/content/read",
		"libre.graph/driveItem/upload/create"
	]
}
```

I guess you want `len(driveItemInvite.Roles) != 0 &&  len(driveItemInvite.LibreGraphPermissionsActions) != 0`

You probably also need to catch the case when both of them are empty.

Reply:
good catch, tanks, missed it, fixed!